### PR TITLE
More feature for ossfs

### PR DESCRIPTION
**Pull request by siyuan0322**

### CODE REVIEW

File: modules/io/python/drivers/io/tests/test_ossfs.py [hunk 1]
```diff
@@ -33,8 +33,7 @@
 import pytest
 import requests
 
-pytestmark = pytest.mark.skipif(True, reason='skip ossfs test')
-
+ossfs = pytest.importorskip("ossfs")
 from ossfs import OSSFileSystem
```
Comment (sighingnow) [inline modules/io/python/drivers/io/tests/test_ossfs.py:37]: The variable name conflicits with the package name...

Comment (siyuan0322) [reply modules/io/python/drivers/io/tests/test_ossfs.py:37]: If import succedd, `ossfs` will be that module, just like `mod = importlib.import_module("mod")`

**Review by sighingnow — APPROVED** [review]
LGTM.

Is the `test_ossfs.py` taken from somewhere?

File: modules/io/python/drivers/io/ossfs.py [hunk 2]
```diff
@@ -394,55 +716,404 @@ def ls(self, path, detail=False, refresh=False, **kwargs):
...
+            version_list = oss2.models.BatchDeleteObjectVersionList()
+            for o in itertools.chain(result.versions, result.delete_marker):
+                version_list.append(oss2.models.BatchDeleteObjectVersion(o.key, o.versionid))
+            if version_list.len() == 0:
+                break
+            self.call_oss(bucket_name, "delete_object_versions", version_list)
+
+            is_truncated = result.is_truncated
+
+            # 查看列举结果是否完整。如果结果不完整，则继续罗列。如果结果已完整，则退出循环。
```
Comment (sighingnow) [inline modules/io/python/drivers/io/ossfs.py:1110]: Comment in english.

**Comment by siyuan0322** [comment]
> LGTM.
> 
> Is the `test_ossfs.py` taken from somewhere?

The architecture and test cases are referred and adapted from [s3fs](https://github.com/dask/s3fs).

**Comment by sighingnow** [comment]
> > LGTM.
> > Is the `test_ossfs.py` taken from somewhere?
> 
> The architecture and test cases are referred and adapted from [s3fs](https://github.com/dask/s3fs).

Leave a note and reference in copyright header please.

If the test file is taken from s3fs, please keep its original copyright header, and add an entry to Acknowledgements (in readme), license, as well as notice.txt.

File: NOTICE.txt [hunk 3]
```diff
@@ -35,3 +35,8 @@ This product includes software from the flingfd project
 This product includes software from the boost/leaf project
  * Copyright (c) 2018-2020 Emil Dotchevski and Reverge Studios, Inc.
  * https://github.com/boostorg/leaf
+
+
```
Comment (sighingnow) [inline NOTICE.txt:39]: Why two empty lines..

File: README.rst [hunk 4]
```diff
@@ -205,6 +205,9 @@ Acknowledgements
 - `flat_hash_map <https://github.com/skarupke/flat_hash_map>`_, an efficient hashmap implementation;
 - `pybind11 <https://github.com/pybind/pybind11>`_, a library for seamless operability between C++11 and Python;
 - `uri <https://github.com/cpp-netlib/uri>`_, a library for URI parsing.
+- `s3fs <https://github.com/dask/s3fs>`_, a library provide a convenient Python filesystem interface for S3.
+
+
```
Comment (sighingnow) [inline README.rst:210]: Why those extra empty lines..

**Review by sighingnow — COMMENTED** [review]
But that's all right~

Thanks for such effort!

**Pull request merged**